src/compiler/comments.ts

```diff
@@ -198,9 +198,9 @@ namespace ts {
             }
 
             // Leading comments are emitted at /*leading comment1 */space/*leading comment*/space
-            emitPos(commentPos);
+            if (emitPos) emitPos(commentPos);

```

Comment:
Source map support is not currently exposed publicly, so a printer created through the public API would not be able to provide an `emitPos` callback. I'll amend the parameter to include `undefined`.